independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InterceptorGenerator.java

```diff
@@ -224,38 +228,70 @@ private void addIntercept(MethodCreator intercept, MethodInfo interceptorMethod,
+                            String.format("Interceptor method %s#%s()", interceptorMethod.declaringClass().name(),
+                                    interceptorMethod.name()));
+                    // Use reflection fallback
+                    ResultHandle paramTypesArray = callbackIndexTrueBranch.newArray(Class.class,
+                            callbackIndexTrueBranch.load(1));
+                    callbackIndexTrueBranch.writeArrayValue(paramTypesArray, 0,
+                            callbackIndexTrueBranch.loadClass(invocationContextClass));
+                    ResultHandle argsArray = callbackIndexTrueBranch.newArray(Object.class, callbackIndexTrueBranch.load(1));
+                    callbackIndexTrueBranch.writeArrayValue(argsArray, 0, invocationContextRH);
+                    reflectionRegistration.registerMethod(interceptorMethod);
+                    ret = callbackIndexTrueBranch.invokeStaticMethod(MethodDescriptors.REFLECTIONS_INVOKE_METHOD,
+                            callbackIndexTrueBranch.loadClass(interceptorMethod.declaringClass()
+                                    .name()
+                                    .toString()),
+                            callbackIndexTrueBranch.load(interceptorMethod.name()), paramTypesArray,
+                            interceptorRH, argsArray);
+                } else {
+                    if (interceptorMethod.declaringClass().toString().equals(providerType.className())) {
+                        ret = callbackIndexTrueBranch.invokeVirtualMethod(
+                                MethodDescriptor.ofMethod(interceptorMethod.declaringClass().toString(),
+                                        interceptorMethod.name(),
+                                        retType,
+                                        invocationContextClass),
+                                interceptorRH, invocationContextRH);
+                    } else {
+                        ResultHandle superInterceptorRH = callbackIndexTrueBranch.checkCast(interceptorRH,
+                                interceptorMethod.declaringClass().toString());
+                        ret = callbackIndexTrueBranch.invokeVirtualMethod(
+                                MethodDescriptor.ofMethod(interceptorMethod.declaringClass().toString(),
+                                        interceptorMethod.name(),
+                                        retType,
+                                        invocationContextClass),
+                                superInterceptorRH, invocationContextRH);
+                    }
+                }
+                i++;
+                callbackIndexTrueBranch
+                        .returnValue(InterceptionType.AROUND_INVOKE.equals(interceptionType) ? ret : trueBranch.loadNull());
             }
-            trueBranch.returnValue(InterceptionType.AROUND_INVOKE.equals(interceptionType) ? ret : trueBranch.loadNull());
+            trueBranch.returnValue(trueBranch.loadNull());
```

Comment:
only add it if interceptorMethods.size() <= 1 because is right method is not called.